storage/pc-ata/Makefile

```diff
@@ -4,13 +4,10 @@
 # Copyright 2018, 2019 Phoenix Systems
 #
 
-$(PREFIX_PROG)pc-ata: $(addprefix $(PREFIX_O)storage/pc-ata/, atadrv.o atasrv.o mbr.o) $(PREFIX_A)libext2.a
+$(PREFIX_PROG)pc-ata: $(addprefix $(PREFIX_O)storage/pc-ata/, ata.o atasrv.o mbr.o) $(PREFIX_A)libext2.a
 	$(LINK)
 
-$(PREFIX_A)libatadrv.a: $(PREFIX_O)storage/pc-ata/atadrv.o
+$(PREFIX_A)libata.a: $(PREFIX_O)storage/pc-ata/ata.o
 	$(ARCH)
 
-$(PREFIX_H)atasrv.h: storage/pc-ata/atasrv.h
-	$(HEADER)
-
-all: $(PREFIX_PROG_STRIPPED)pc-ata $(PREFIX_A)libatadrv.a $(PREFIX_H)atasrv.h
+all: $(PREFIX_PROG_STRIPPED)pc-ata $(PREFIX_A)libata.a
```

Comment:
Why we don't want to install header?

Reply:
We have a circular dependency. Ext2 filesystem depends on PC-ATA (atasrv_read and atasrv_write to be specific). On the other hand PC-ATA driver depends on the filesystem libraries (in our case on libext2.a). The default build order is building filesystems first (PC-ATA makefile haven't triggered yet, the atasrv header is not installed), then the drivers.

Reply:
Ok. I think that ext2 should not depend on pc-ata or any other storage driver (at least that should be our goal). Ext2 should get read/write callbacks on initialization and use them without knowledge of any particular hardware. @pawelpisarczyk do you agree with me on that?

Reply:
Yes